tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -165,6 +173,16 @@ object NativeConfig {
     def withOptimize(value: Boolean): NativeConfig =
       copy(optimize = value)
 
+    def withMultithreadingSupport(enabled: Boolean): NativeConfig = {
+      copy(multithreadingSupport = enabled)
+        .withLinktimeProperties(
+          linktimeProperties.updated(
+            "scala.scalanative.meta.linktimeinfo.isMultithreadingEnabled",
+            enabled
+          )
+        )
+    }
```

Comment:
I think the setting should be defined in one place instead of two.
What happens if I set:
```scala
nativeConfig ~= {_.withLinktimeProperties(
  linktimeProperties.updated(
    "scala.scalanative.meta.linktimeinfo.isMultithreadingEnabled",true)
  )
}
```
?
Will this enable "half" of the multithreading code (just the runtime one without the compiler)

And if I do:
```scala
nativeConfig ~= {
  _.withMultithreadingSupport(true)
    .withLinktimeProperties(linktimeProperties - "scala.scalanative.meta.linktimeinfo.isMultithreadingEnabled")
}
```

It will have only the compiler support without the runtime.

I think we should define only a single setting that can be enabled/disabled atomically.

Reply:
That was a good point, I've replaced `isMultithreadingEnabled` and `isWindows` properties with predefined values (computed from NativeConfig or Platform settings) and introduced custom properties that can be set by the user. At this point `Discover.linktimeProperties` might be obsolete, however, I guess we can keep it for future uses, eg. reading props from env variable - right now introducing parsing would be quite problematic.

Reply:
You can also see the direction I am taking is to use `isWindows` internal to tools so the tools do the right thing.